crates/bevy_color/src/hsla.rs

```diff
@@ -0,0 +1,223 @@
+use crate::{Alpha, LinearRgba, Luminance, Mix, Srgba};
+use bevy_reflect::{Reflect, ReflectDeserialize, ReflectSerialize};
+use bevy_render::color::{Color, HslRepresentation};
+use serde::{Deserialize, Serialize};
+
+/// Color in Hue-Saturation-Lightness color space with alpha
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize, Reflect)]
+#[reflect(PartialEq, Serialize, Deserialize)]
+pub struct Hsla {
+    /// The hue channel. [0.0, 360.0]
+    pub hue: f32,
+    /// The saturation channel. [0.0, 1.0]
+    pub saturation: f32,
+    /// The lightness channel. [0.0, 1.0]
+    pub lightness: f32,
+    /// The alpha channel. [0.0, 1.0]
+    pub alpha: f32,
+}
+
+impl Hsla {
+    /// Construct a new [`Hsla`] color from components.
+    ///
+    /// # Arguments
+    ///
+    /// * `hue` - Hue channel. [0.0, 360.0]
+    /// * `saturation` - Saturation channel. [0.0, 1.0]
+    /// * `lightness` - Lightness channel. [0.0, 1.0]
+    /// * `alpha` - Alpha channel. [0.0, 1.0]
+    pub const fn new(hue: f32, saturation: f32, lightness: f32, alpha: f32) -> Self {
+        Self {
+            hue,
+            saturation,
+            lightness,
+            alpha,
+        }
+    }
+
+    /// Construct a new [`Hsla`] color from (h, s, l) components, with the default alpha (1.0).
+    ///
+    /// # Arguments
+    ///
+    /// * `hue` - Hue channel. [0.0, 360.0]
+    /// * `saturation` - Saturation channel. [0.0, 1.0]
+    /// * `lightness` - Lightness channel. [0.0, 1.0]
+    pub const fn hsl(hue: f32, saturation: f32, lightness: f32) -> Self {
+        Self::new(hue, saturation, lightness, 1.0)
+    }
+}
+
+impl Default for Hsla {
+    fn default() -> Self {
+        Self::new(0., 0., 0., 1.)
```

Comment:
Possible consistency issue: some colours default to white (`Color`, `Srgba`, `LinearRgba`), while others default to black (`Hsla`, `Lcha`, `Oklaba`). I don't mind which is the default, but it should probably be consistent.

Reply:
Done